### ISSUE TITLE
Meta Meta Programming

### ISSUE THREAD

**Issue by sirinath** [kframework/k-legacy#220]
Is it possible to have a layer on top of K which would let users use more custom syntax for like in Campl, Lisp, etc. Event the ability to mix and match each scheme and also import work from other frameworks.

**Comment by adamsmd** [comment]
I do not understand what you are asking.  Could you give a concrete example?

**Comment by sirinath** [comment]
Let me put it in a different way. Say I define a language though Whole, Rascal, Camlp5, etc. and want to re use this within K. Then I should be able to define a way to retrofit the K to handle this syntax through macros and meta programming.

Similarly you can use K to define macros which can simplify K development by re using common definitions or providing concise sugar to define certain elements. Also ability to use schemes which the user is currently used to like using and alternate syntax similar to Lips, Rascal Camlp, etc.

In other words you have layers:
Base K
Layer 1 of k macros using base K
Layer 2 of K macros using base K and Layer 1 macros
....
....

**Comment by sirinath** [comment]
This way the new K framework can be generated from the old K framework

**Comment by dwightguth** [comment]
Closing. If you still want an answer, reopen and we will try to respond. I am really sorry we didn't do so before, but the project ops structure was in a very bad state and we are only now going through everything and trying to set up a system to make sure all questions are answered.
